vcs/gitcmd/repo.go

```diff
@@ -221,6 +235,30 @@ func (r *Repository) Branches(opt vcs.BranchesOptions) ([]*vcs.Branch, error) {
 	return branches, nil
 }
 
+func (r *Repository) branchesWithCommit(commitID string) (map[string]bool, error) {
+	cmd := exec.Command("git", "branch", "--contains", commitID)
```

Comment:
For security/paranoia, either use `checkArgSpecSafety` on `commitID`, or use the following line instead:

```go
cmd := exec.Command("git", "branch", "--contains="+commitID)
```

The latter is better IMO.